src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,238 @@
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getUserId());
+            statement.executeUpdate();
+            deleteUserFromUsersRoles(user.getUserId());
+            insertToUsersRoles(user);
+            return user;
+        } catch (SQLException e) {
+            throw new DataProcessingException(e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM users WHERE user_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            deleteUserOrdersProducts(id);
+            deleteUserOrders(id);
+            deleteUserShoppingCartProducts(id);
+            deleteUserShoppingCart(id);
+            deleteUserFromUsersRoles(id);
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(e);
+        }
+    }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String query = "SELECT * FROM users WHERE login = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                User user = createUserFromResultSetAndSetRoles(resultSet);
+                return Optional.of(user);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(e);
+        }
+        return Optional.empty();
+    }
+
+    private void selectAndInsertRelativeRole(User user) throws SQLException {
+        String insertQuery = "INSERT INTO users_roles (user_id, role_id) VALUES (?, ?)";
+        String selectQuery = "SELECT role_id FROM roles WHERE role_name = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Role role : user.getRoles()) {
+                PreparedStatement selectStatement =
+                        connection.prepareStatement(selectQuery);
+                selectStatement.setString(1, role.getRoleName().name());
+                ResultSet resultSet = selectStatement.executeQuery();
+                resultSet.next();
+                PreparedStatement insertStatement =
+                        connection.prepareStatement(insertQuery);
+                insertStatement.setLong(1, user.getUserId());
+                insertStatement.setLong(2, resultSet.getLong("role_id"));
+                insertStatement.executeUpdate();
+            }
+        }
+    }
+
+    private User createUserFromResultSetAndSetRoles(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private User getUserFromResultSet(ResultSet resultSet) throws SQLException {
```
setting roles it's part of fetching user, so we don't have to ad this to method name